### PR TITLE
Add a test that lays out strings visually

### PR DESCRIPTION
**Pull request by swankjesse**
Testing flexbox should be easy. With this its a picnic.

### CODE REVIEW

File: redwood-flexbox-engine/src/commonTest/kotlin/app/cash/redwood/FlexboxTest.kt [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+ */
+package app.cash.redwood
+
+import app.cash.redwood.FlexDirection.Companion.Row
+import app.cash.redwood.FlexDirection.Companion.RowReverse
+import kotlin.test.Test
+import kotlin.test.assertEquals
+
+class FlexboxTest {
+  private val imdbTop4 = listOf(
```
Comment (swankjesse) [inline redwood-flexbox-engine/src/commonTest/kotlin/app/cash/redwood/FlexboxTest.kt:24]: Definitely open to suggestions for another theme here. I want 3-5 short phrases with variable word length.

File: redwood-flexbox-engine/src/commonTest/kotlin/app/cash/redwood/FlexboxTest.kt [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+ * limitations under the License.
+ */
+package app.cash.redwood
+
+import app.cash.redwood.FlexDirection.Companion.Row
+import app.cash.redwood.FlexDirection.Companion.RowReverse
+import kotlin.test.Test
+import kotlin.test.assertEquals
+
+class FlexboxTest {
```
Comment (swankjesse) [inline redwood-flexbox-engine/src/commonTest/kotlin/app/cash/redwood/FlexboxTest.kt:23]: This test isn’t exhaustive! But I think it’s more real than the FakeNode one we inherited! In particular this is starting to prove out which parameters are inputs and which ones are outputs.

File: redwood-flexbox-engine/src/commonTest/kotlin/app/cash/redwood/FlexboxTest.kt [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+
+  private fun FlexboxEngine.layout(
+    width: Int,
+    height: Int,
+  ): String {
+    val canvas = StringCanvas(width, height)
+    val widthSpec = MeasureSpec.from(width, MeasureSpecMode.Exactly)
+    val heightSpec = MeasureSpec.from(height, MeasureSpecMode.Exactly)
+
+    flexLines = when (flexDirection) {
```
Comment (swankjesse) [inline redwood-flexbox-engine/src/commonTest/kotlin/app/cash/redwood/FlexboxTest.kt:144]: This mutates the FlexBoxEngine. I don’t think that requiring callers to do this is a good API. We should fix that. Also callers shouldn’t need to interrogate the flex direction to call the right function!

**Review by JakeWharton — APPROVED** [review]
Probably want to rebase on #383, although may want to wait until its mutable property follow-up.

**Comment by swankjesse** [comment]
Rebased!

**Review by colinrtwhite — APPROVED** [review]
This is awesome 🙏

**Pull request merged**